wolfcrypt/src/pkcs7.c

```diff
@@ -3541,8 +3582,17 @@ static int PKCS7_VerifySignedData(PKCS7* pkcs7, const byte* hashBuf,
                 idx = localIdx;
             }
             else {
-                if (!degenerate && ret != 0)
+
+                /* if pkcs7->content and pkcs7->contentSz are set, try to
+                   process as a detached signature */
+                if (!degenerate &&
+                    (pkcs7->content != NULL && pkcs7->contentSz != 0)) {
+                    detached = 1;
+                }
+
+                if (!degenerate && !detached && ret != 0)
```

Comment:
It would be nice to error out early here. If pkcs7->detached is not set yet detached variable is set and it is not a degenerate

Reply:
I believe this already errors out directly by breaking out of the switch() statement with ret != 0.